impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -38,65 +36,54 @@
 import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxReader;
 import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxWriter;
 
-import static org.apache.maven.impl.ImplUtils.nonNull;
 import static org.apache.maven.impl.StaxLocation.getLocation;
 import static org.apache.maven.impl.StaxLocation.getMessage;
 
 @Named
 @Singleton
 public class DefaultPluginXmlFactory implements PluginXmlFactory {
+
+    private static final PluginDescriptorStaxWriter WRITER = new PluginDescriptorStaxWriter();
+    private static final PluginDescriptorStaxReader READER = new PluginDescriptorStaxReader();
+
     @Override
     public PluginDescriptor read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
-        nonNull(request, "request");
-        Path path = request.getPath();
-        URL url = request.getURL();
-        Reader reader = request.getReader();
-        InputStream inputStream = request.getInputStream();
-        if (path == null && url == null && reader == null && inputStream == null) {
-            throw new IllegalArgumentException("path, url, reader or inputStream must be non null");
-        }
+        READER.setAddDefaultEntities(request.isAddDefaultEntities());
```

Comment:
This cannot be threadsafe, and that's one reason why they are not singletons.

Reply:
yes wanted to challenge this. Will test. Thanks

Reply:
> Will test

unable to on this lvl. need higher test for this. Powermock is outdated considering not an option. 

How to test it?